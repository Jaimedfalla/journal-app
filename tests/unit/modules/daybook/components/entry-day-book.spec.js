import { shallowMount } from "@vue/test-utils";
import EntryDayBook from "@/modules/daybook/components/EntryDayBook.vue";
import { entry } from "../__mocks__/test-journal-state";

describe('Pruebas en Entry component',()=>{

    let wrapper;
    const mockRouter = {
        push: jest.fn()
    }

    beforeAll(()=>{
        wrapper = shallowMount(EntryDayBook,{
            props:{
                entry
            },
            global:{
                mocks:{
                    $router:mockRouter
                }
            }
        })
    })

    it('Debe hacer match con el snapshot',()=>{
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('Debe redireccionar al hacer click en el entry-container',()=>{
        wrapper.find('.entry-container').trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name:'entry',params:{id:entry.id}})
    })

    it('Pruebas en las propiedades computadas',()=>{
        expect(wrapper.vm.day).toBe(9)
        expect(wrapper.vm.month).toBe("Enero")
        expect(wrapper.vm.yearDay).toBe("2023, Lunes")
    })
})