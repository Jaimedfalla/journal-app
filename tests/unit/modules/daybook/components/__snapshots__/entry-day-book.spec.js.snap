// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry component Debe hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <!-- Para redireccionar a otra vista -->
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">9</span><span class="mx-1 fs-5">Enero</span><span class="mx-2 fw-light">2023, Lunes</span></div>
  <div class="entry-description">Prueba de update Entry con foto</div>
</div>
`;
